app/src/org/commcare/android/api/ExternalApiReceiver.java

```diff
@@ -149,18 +149,8 @@ private void performAction(final Context context, Bundle b) {
             tryLocalLogin(context, username, password);
         } else if(b.getString("commcareaction").equals("sync")) {
             
-            boolean formsToSend = checkAndStartUnsentTask(context, new ProcessTaskListener() {
+            boolean formsToSend = checkAndStartUnsentTask(context);
 
-                public void processTaskAllProcessed() {
-                    //Don't cancel the dialog, we need it to stay in the foreground to ensure things are set
-                }
-                
-                public void processAndSendFinished(int result, int successfulSends) {

```

Comment:
This listeners was never used by checkAndStartUnsent, so I removed it as an argument.